backend/security_utils.py

```
@@ -66,51 +67,73 @@ def validate_port(port: int) -> int:
+        # 4. Check for absolute paths (both Unix and Windows style) - block by default for security
+        if path.startswith('/') or (len(path) > 1 and path[1] == ':') or path.startswith('\\'):
+            if not allow_absolute:
+                raise SecurityValidationError(f"Path contains directory traversal attempt: {original_path}")
+
+        # 5. Normalize the path to resolve '..' and '.' components
         normalized_path = os.path.normpath(path)
         
-        # If base_path is provided, ensure the path is within it using cross-platform approach
+        # 6. Check for directory traversal patterns AFTER normalization
+        if '..' in normalized_path.split(os.sep):
+             raise SecurityValidationError(f"Path contains directory traversal attempt: {original_path}")
```

Flake8 (7.2.0)

[error] indentation is not a multiple of 4

(E111)